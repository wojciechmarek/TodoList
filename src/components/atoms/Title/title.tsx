import { TitleProps } from './title.interface';
import { TitleStyle } from './text.styled';

export const Title = (props: TitleProps) => {
  const { title, ...params } = props;

  return <TitleStyle {...params}>{title}</TitleStyle>;
};
