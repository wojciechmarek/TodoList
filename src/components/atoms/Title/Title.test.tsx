import { configure, render } from '@testing-library/react';
import { Title } from './title';
import { TitleProps } from './title.interface';
import '@testing-library/jest-dom';
import TestRenderer from 'react-test-renderer';

describe('Title', () => {
  let titleProps: TitleProps;

  beforeEach(() => {
    configure({ throwSuggestions: true });

    titleProps = {
      title: 'hello title world',
    };
  });

  it('should render', () => {
    const { baseElement } = render(<Title {...titleProps} />);
    expect(baseElement).toBeTruthy();
  });

  it('should match the snapshot', () => {
    const tree = TestRenderer.create(<Title {...titleProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('when the title is rendered', () => {
    it('should have the correct value', () => {
      const { getByRole } = render(<Title {...titleProps} />);
      const title = getByRole('heading', { name: /hello title world/i });
      expect(title).toBeInTheDocument();
    });
  });
});
