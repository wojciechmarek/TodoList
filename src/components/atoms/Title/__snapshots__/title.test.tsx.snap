// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should match the snapshot 1`] = `
<h1
  className="emotion-0"
>
  hello title world
</h1>
`;
